pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -87,10 +87,15 @@ func ApplyFeatureGates() {
 		klog.Infof("TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory")
 	}
 
-	// Only register EvenPodsSpreadPredicate if the feature is enabled
+	// Only register EvenPodsSpread predicate & priority if the feature is enabled
 	if utilfeature.DefaultFeatureGate.Enabled(features.EvenPodsSpread) {
+		klog.Infof("Registering EvenPodsSpread predicate and priority function")
+		// register predicate
 		factory.InsertPredicateKeyToAlgorithmProviderMap(predicates.EvenPodsSpreadPred)
 		factory.RegisterFitPredicate(predicates.EvenPodsSpreadPred, predicates.EvenPodsSpreadPredicate)
+		// register priority
+		factory.InsertPriorityKeyToAlgorithmProviderMap(priorities.EvenPodsSpreadPriority)
+		factory.RegisterPriorityFunction(priorities.EvenPodsSpreadPriority, priorities.CalculateEvenPodsSpreadPriority, 1)
```

Comment:
This function has already been deprecated, we should use map-reduce pattern instead of this. I created https://github.com/kubernetes/kubernetes/pull/80225 to remove this from the codebase.

> // RegisterPriorityFunction registers a priority function with the algorithm registry. Returns the name,
// with which the function was registered.
// DEPRECATED
// Use Map-Reduce pattern for priority functions.

Reply:
Friendly reminder on the usage of deprecated `RegisterPriorityFunction`. :)

Reply:
#80225 needs further considerations. I will take another round of review later.